exercises/matthew_bowles/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,188 @@
+#include "main.h"
+
+using namespace std;
+
+int main(int argc, char *argv[])
+{
+	if (argc < 2)
+	{
+		ofstream output;
+		output.open("Output.txt");
+		output << "Missing input file argument" << '\n';
+		return 1;
+	}
+	map<string, int> words = loadFile(argv[1]);
+	vector<pair<string, int>> sortedWords = sortWords(words);
+	outputCount(sortedWords);
+	return 0;
+}
+
+
+/** Loads word input from a file into a map 
+*
+* @param path The path to the input file
+* @return A map of words and their number of uses
+*
+*/
+map<string, int> loadFile(const string& path)
+{
+	map<string, int> words;
+	ifstream fileIn;
+	string line;
+	string word;
+	
+	fileIn.open(path);
+	if (fileIn.is_open())
+	{
+		while (getline(fileIn, line))
+		{
+			splitLine(line, words);
+		}
+	}
+	else cout << "unable to open file";
+	return words;
+}
+
+/** Splits an input line into seperate words
+*
+* @param line The line to split
+* @param words The map of words to hold the resulting words
+*
+*/
+void splitLine(string& line, map<string, int>& words)
+{
+	string word;
+	size_t space;
+
+	space = line.find(" ");
+	while (space != string::npos)
+	{
+		word = line.substr(0, space);
+		++space;
+		line = line.substr(space);
+		addWord(words, word);
+		space = line.find(" ");
+	}
+	addWord(words, line);
+}
+
+/** Adds a word to a given map if applicable
+*
+* @param words The map of words to add to
+* @param word The word to be added
+*
+*/
+void addWord(map<string, int>& words, string& word)
+{
+	map<string, int>::const_iterator existing;
```

Comment:
Don't define variables in a scope that they are not required in yet (unless there is a good reason)